lib/ngtcp2_conn.c

```diff
@@ -4702,7 +4702,11 @@ static int conn_recv_handshake_cpkt(ngtcp2_conn *conn, const ngtcp2_path *path,
       if (nread == NGTCP2_ERR_DISCARD_PKT) {
         goto fin;
       }
-      if (nread != NGTCP2_ERR_CRYPTO && (pkt[0] & NGTCP2_HEADER_FORM_BIT) &&
+      assert(pktlen > 1);
+      if (nread != NGTCP2_ERR_CRYPTO &&
+          (pkt[0] & NGTCP2_HEADER_FORM_BIT) &&
```

Comment:
I think it would be safer to check that pktlen > 4 holds here. pktlen might be 1.

Reply:
Good point. Fixed.

Reply:
Isn't it better to add pktlen > 4 in if condition rather than assert?  assertion may fail and it crashes application.

Reply:
I originally chose to assert because I expect conn_recv_handshake_pkt() to return NGTCP2_ERR_DISCARD_PKT if pktlen <= 5 because of the check in ngtcp2_pkt_decode_hd_long(). Therefore I assumed that if we reach this point with pktlen <= 4, it's necessarily a bug.

I now moved the check to the condition as suggested.

Reply:
Thank you for explanation.  I think the assumption is ok for the current code.  Sorry I haven't checked the function.